src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -226,8 +197,8 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
       } finally {
         data.releaseReadLock();
       }
-      this.data = dataCopy;
-      s_battleCalculator.setGameData(dataCopy);
+      ProData.setData(dataCopy);
```

Comment:
So that may be the case, be aware though that the call to `cloneGameData` is a performance killer when a game gets a bit larger. For example, a round 13 world at war game can take a second or more to write to disk and read it back. Really the fix should probably be in `cloneGameData` to do this in memory.